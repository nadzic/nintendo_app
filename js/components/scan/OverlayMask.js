import React, { Component } from 'react';
import { View, StyleSheet, Dimensions } from 'react-native';
import Loader from 'nintendoapp/js/components/common/Loader';

/* eslint-disable global-require */

class OverlayMask extends Component {
  render() {
    const { loading } = this.props;
    const { height, width } = Dimensions.get('window');
    const maskRowHeight = Math.round((height - 200) / 20);
    const maskColWidth = (width - 200) / 2;

    return (
      <View style={styles.maskOutter}>
        <View style={[{ flex: maskRowHeight }, styles.maskRow, styles.maskFrame]} />
        <View style={[{ flex: 30 }, styles.maskCenter]}>
          <View style={[{ width: maskColWidth }, styles.maskFrame]} />
          <View style={styles.maskInner} />
          <View style={[{ width: maskColWidth }, styles.maskFrame]} />
        </View>
        <View style={[{ flex: maskRowHeight }, styles.maskRow, styles.maskFrame]} />
        {loading && <Loader white />}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  maskOutter: {
    position: 'absolute',
    left: 0,
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  maskInner: {
    width: 300,
    backgroundColor: 'transparent',
  },
  maskFrame: {
    backgroundColor: 'rgba(1,1,1,0.6)',
  },
  maskRow: {
    width: '100%',
  },
  maskCenter: {
    flexDirection: 'row',
  },
});

export default OverlayMask;
