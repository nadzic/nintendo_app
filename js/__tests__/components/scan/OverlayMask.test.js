import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import OverlayMask from 'nintendoapp/js/components/scan/OverlayMask';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
  };

  const wrapper = shallow(<OverlayMask {...props} />);

  return {
    props,
    wrapper,
  };
}

describe('OverlayMask component', () => {
  it('should renders correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
