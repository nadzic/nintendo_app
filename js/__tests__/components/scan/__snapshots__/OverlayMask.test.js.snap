// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlayMask component should renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "space-around",
      "left": 0,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 57,
        },
        Object {
          "width": "100%",
        },
        Object {
          "backgroundColor": "rgba(1,1,1,0.6)",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 30,
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": 275,
          },
          Object {
            "backgroundColor": "rgba(1,1,1,0.6)",
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "width": 300,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "width": 275,
          },
          Object {
            "backgroundColor": "rgba(1,1,1,0.6)",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 57,
        },
        Object {
          "width": "100%",
        },
        Object {
          "backgroundColor": "rgba(1,1,1,0.6)",
        },
      ]
    }
  />
</View>
`;
